packages/test/test-end-to-end-tests/src/test/gc/gcStateResetInSummaries.spec.ts

```diff
@@ -0,0 +1,276 @@
+
+    /** Creates a new container with the GC enabled / disabled as per gcAllowed param. */
+    const createContainer = async (gcAllowed?: boolean): Promise<IContainer> => {
+        const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+            dataObjectFactory,
+            [
+                [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+            ],
+            undefined,
+            undefined,
+            { ...defaultRuntimeOptions, gcOptions: { gcAllowed } },
+        );
+        return provider.createContainer(runtimeFactory);
+    };
+
+    /** Loads a summarizer client with the given version (if any). Also enables / disables GC as per disableGC param. */
+    const getNewSummarizer = async (disableGC: boolean, gcAllowed?: boolean, summaryVersion?: string) => {
+        const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+            dataObjectFactory,
+            [
+                [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+            ],
+            undefined,
+            undefined,
+            { ...defaultRuntimeOptions, gcOptions: { gcAllowed, disableGC } },
+        );
+        return loadSummarizer(
+            provider, runtimeFactory, mainContainer.deltaManager.lastSequenceNumber, summaryVersion);
+    };
+
+    /**
+     * Callback that will be called by the document storage service whenever a summary is uploaded by the client.
+     * Update the summary context to include the summary proposal and ack handle as per the latest ack for the
+     * document.
+     */
+    function uploadSummaryCb(summaryTree: ISummaryTree, context: ISummaryContext): ISummaryContext {
+        latestUploadedSummary = summaryTree;
+        latestSummaryContext = context;
+        const newSummaryContext = { ...context };
+        // If we received an ack for this document, update the summary context with its information. The
+        // server rejects the summary if it doesn't have the proposal and ack handle of the previous
+        // summary.
+        if (latestAckedSummary !== undefined) {
+            newSummaryContext.ackHandle = latestAckedSummary.summaryAck.contents.handle;
+            newSummaryContext.proposalHandle = latestAckedSummary.summaryOp.contents.handle;
+        }
+        return newSummaryContext;
+    }
+
+    /** Generates a summary and returns the data store channel sub-tree */
+    async function getSummaryChannelsTree(
+        summarizerClient: { containerRuntime: ContainerRuntime, summaryCollection: SummaryCollection },
+    ) {
+        const summaryResult = await submitAndAckSummary(provider, summarizerClient, logger, true /* fullTree */);
+        latestAckedSummary = summaryResult.ackedSummary;
+
+        assert(
+            latestSummaryContext && latestSummaryContext.referenceSequenceNumber >= summaryResult.summarySequenceNumber,
+            `Did not get expected summary. Expected: ${summaryResult.summarySequenceNumber}. ` +
```

Comment:
#nit why not just do an `assert.equal` here in case `latestSummaryContext` is truthy?